server/game/cards/02_SHD/events/CalculatedLethality.ts

```diff
@@ -0,0 +1,41 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { EventCard } from '../../../core/card/EventCard';
+import { RelativePlayer, WildcardCardType } from '../../../core/Constants';
+
+export default class CalculatedLethality extends EventCard {
+    protected override getImplementationId() {
+        return {
+            id: '0302968596',
+            internalName: 'calculated-lethality',
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.setEventAbility({
+            title: 'Defeat a non-leader unit that costs 3 or less',
+            targetResolver: {
+                cardCondition: (card) => card.isNonLeaderUnit() && card.cost <= 3,
+                immediateEffect: AbilityHelper.immediateEffects.sequential([
+                    // TODO: this is a hack to store data before it's defeated.
+                    // once last known state is implemented, we need to read data from that
+                    AbilityHelper.immediateEffects.handler((context) => ({
+                        handler: () => context.targets.upgradeAmount = context.target.upgrades.length
+                    })),
+                    AbilityHelper.immediateEffects.defeat()
+                ])
+            },
+            ifYouDo: (ifYouDoContext) => ({
```

Comment:
This actually should be a "then" not an "ifYouDo" because of this:

![image](https://github.com/user-attachments/assets/c4f72a8b-1a4e-4cfe-9f8a-f556df5c048b)

https://nexus.cascadegames.com/resources/Rules_Clarifications/